mobile/src/employee-details/employee-details-events-list.tsx

```diff
@@ -1,125 +1,116 @@
+import { CalendarEvent } from '../reducers/calendar/calendar-event.model';
 import { EventManagementToolset } from './event-management-toolset';
-import { Employee } from '../reducers/organization/employee.model';
 import { CalendarEventIcon } from '../calendar/calendar-event-icon';
 import { Nullable } from 'types';
+import { EventActionsContainer } from './event-action-provider';
+import { List } from 'immutable';
 
+//============================================================================
 interface EmployeeDetailsEventsListProps {
-    events: Map<Employee, CalendarEvent[]>;
-    eventSetNewStatusAction: (employeeId: string, calendarEvent: CalendarEvent, status: CalendarEventStatus) => void;
+    eventLogicContainers: EventActionsContainer[];
     hoursToIntervalTitle: (startWorkingHour: number, finishWorkingHour: number) => Nullable<string>;
     showUserAvatar?: boolean;
-    canApprove: boolean;
-    canReject: boolean;
-}
-
-interface EmployeeDetailItem {
-    employee: Employee;
-    calendarEvent: CalendarEvent;
 }
 
+//============================================================================
 export class EmployeeDetailsEventsList extends Component<EmployeeDetailsEventsListProps> {
     private readonly eventDigitsDateFormat = 'DD/MM/YYYY';
 
+    //----------------------------------------------------------------------------
     public render() {
 
         const events = this.prepareEvents();
 
         return (<FlatList
-                    scrollEnabled={false}
-                    data={events}
-                    keyExtractor={this.keyExtractor}
-                    renderItem={this.renderItem} />
+                scrollEnabled={false}
+                data={events}
+                keyExtractor={this.keyExtractor}
+                renderItem={this.renderItem}/>
         );
     }
 
-    private prepareEvents(): Nullable<EmployeeDetailItem[]> {
-        const { events } = this.props;
+    //----------------------------------------------------------------------------
+    private prepareEvents(): Nullable<EventActionsContainer[]> {
+        const { eventLogicContainers } = this.props;
 
-        if (!events) {
+        if (!eventLogicContainers) {
             return null;
         }
 
-        return events
-            .sortBy((_, employee) => employee.name)
-            .map((calendarEvents, employee) =>
-                Set(calendarEvents)
-                    .sort((a, b) => b.dates.startDate.valueOf() - a.dates.startDate.valueOf())
-                    .map(calendarEvent => ({
-                        employee: employee,
-                        calendarEvent: calendarEvent
-                    } as EmployeeDetailItem))
-            )
-            .flatten()
-            .toIndexedSeq()
-            .toArray();
+        return List(eventLogicContainers).sort((left, right) => {
+            const nameCompareResult = left.employee.name.localeCompare(right.employee.name);
```

Comment:
Don't you want to add a separate event action comparator?

Reply:
Done. 1a96a70f449eca26d5d8752579a221d87c2ad3ce